src/toxcore/dht/server/mod.rs

```diff
@@ -422,17 +429,43 @@ impl Server {
             )
         }
 
+        let mut futures = Vec::new();
+
+        if self.is_ipv6_enabled {
+            if let Some(v6) = target_peer.assoc6.saddr {
```

Comment:
Try to use https://doc.rust-lang.org/std/option/enum.Option.html#method.and_then instead of `if let Some`